Automattic/jetpack
# Fix Boost plugin linting issues

Following the [integration of Jetpack Boost](https://github.com/Automattic/jetpack/pull/20512) into this repo and the update of the [eslint-plugin-jsdoc](https://github.com/Automattic/jetpack/pull/20515) I tested and reviewed the JS linting process and results and I am proposing these following changes. Please note that all the changes proposed might not be correct as I am still getting acquainted with this new repo setup.

#### Changes proposed in this Pull Request:

*  Update the `.eslintignore` at the root of the repo as some of the paths/files related to the Boost plugin were incorrect. I have learnt today also that the `.eslintignore` file at the root of the project is not taken in consideration by a eslint process running deeper down the tree (Am I understanding [eslint documentation](https://eslint.org/docs/user-guide/configuring/ignoring-code#the-eslintignore-file) correctly?). I am assuming this `.eslintignore` is required by the root eslint process and the ignore rules are used to prevent unwanted errors when scanning in the Boost plugin directory. Also as a note locally the root linting process (I ran `pnpm run lint` is returning a lot of errors from a bit everywhere in the repo codebase.
* in the `.eslintrc.js` file of the Boost plugin this PR does: 
     * Replace the `'../../../.eslintrc.react.js',`extends entry by `../../../.eslintrc.js`  as Boost doest not use React.
     * Add some more ignore rules to avoid some unwanted errors as the rules in the `.eslintignore` root of the repo is not taken in consideration (see above)
     * Add a rule to check for incorrect text domains. It allowed me to fix a missing one :) 
* Fix a path in the `tsconfig.json` file. Though I am not sure it makes a difference. Maybe `dist` meant any dist directories. Anyway I made it more explicit.
- Fix linting issues in `.ts` files. Most of them only related to JSDoc because of the recent [eslint-plugin-jsdoc](https://github.com/Automattic/jetpack/pull/20515) update. Also fix one missing text domain.
- Remove 2 files which in which the code is unused in the Boost codebase.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
n/a

#### Testing instructions:
- Go to the `projects/plugins/boost` folder. Run `pnpm run lint:js`. The process should complete successfully.
